### PR TITLE
Var to nc var (replaces #2791)

### PR DESCRIPTION
**Pull request by infotroph**
Incorporates @moki1202's work from #2791, which is ready to merge but would be a pain to do because the upstream branch is gone. Opening this PR as the easiest way forward.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [ ] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of .zenodo.json
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: models/gday/R/model2netcdf.GDAY.R [hunk 1]
```diff
@@ -87,32 +87,32 @@ model2netcdf.GDAY <- function(outdir, sitelat, sitelon, start_date, end_date) {
...
-    #var[[12]]  <- PEcAn.utils::to_ncvar("LitterBiomass", "kgC/m2", list(lon,lat,t), -999)
-    #var[[13]]  <- PEcAn.utils::to_ncvar("SoilC", "kgC/m2", list(lon,lat,t), -999)
+    #nc_var[[6]]  <- PEcAn.utils::to_ncvar("LeafLitter", "kgC/m2/s", list(lon,lat,t), -999 )
+    #nc_var[[7]]  <- PEcAn.utils::to_ncvar("WoodyLitter", "kgC/m2/s", list(lon,lat,t), -999)
+    #nc_var[[8]]  <- PEcAn.utils::to_ncvar("RootLitter", "kgC/m2/s", list(lon,lat,t), -999)
+    #nc_var[[9]]  <- PEcAn.utils::to_ncvar("LeafBiomass", "kgC/m2", list(lon,lat,t), -999)
+    #nc_var[[10]]  <- PEcAn.utils::to_ncvar("WoodBiomass", "kgC/m2", list(lon,lat,t), -999)
+    #nc_var[[11]]  <- PEcAn.utils::to_ncvar("RootBiomass", "kgC/m2", list(lon,lat,t), -999)
+    #nc_var[[12]]  <- PEcAn.utils::to_ncvar("LitterBiomass", "kgC/m2", list(lon,lat,t), -999)
+    #nc_var[[13]]  <- PEcAn.utils::to_ncvar("SoilC", "kgC/m2", list(lon,lat,t), -999)
```
Comment (mdietze) [inline models/gday/R/model2netcdf.GDAY.R:115]: If you change the name of this list from var to nc_var, you also need to change where the list is used in the nc_create, for, and ncvar_put below.

File: models/linkages/R/model2netcdf.LINKAGES.R [hunk 2]
```diff
@@ -97,23 +97,23 @@ model2netcdf.LINKAGES <- function(outdir, sitelat, sitelon, start_date = NULL,
...
+    nc_var[[7]]  <- ncdf4::ncvar_def("HeteroResp", "kgC/m2/s", list(dim.lat, dim.lon, dim.t), -999)
+    nc_var[[8]]  <- ncdf4::ncvar_def("NPP", "kgC/m2/s", list(dim.lat, dim.lon, dim.t), -999)
+    nc_var[[9]]  <- ncdf4::ncvar_def("NEE", "kgC/m2/s", list(dim.lat, dim.lon, dim.t), -999)
+    nc_var[[10]] <- ncdf4::ncvar_def("Evap", "kg/m2/s", list(dim.lat, dim.lon, dim.t), -999)
+    nc_var[[11]] <- ncdf4::ncvar_def("AGB.pft", "kgC/m2", list(dim.lat, dim.lon, dim.t, dim.pfts), -999)
+    nc_var[[12]] <- ncdf4::ncvar_def("Fcomp", "kgC/kgC", list(dim.lat, dim.lon, dim.t, dim.pfts), -999)
+    nc_var[[13]] <- ncdf4::ncvar_def("LAI", "m2/m2", list(dim.lat, dim.lon, dim.t), -999)
+    nc_var[[14]] <- ncdf4::ncvar_def("SoilMoist", "m2/m2", list(dim.lat, dim.lon, dim.t), -999)
+    nc_var[[15]] <- ncdf4::ncvar_def("AbvGrndWood", "kgC/m2", list(dim.lat, dim.lon, dim.t), -999)
+    nc_var[[16]] <- ncdf4::ncvar_def("PFT", units = "", dim = list(dim.pfts),
```
Comment (mdietze) [inline models/linkages/R/model2netcdf.LINKAGES.R:116]: same point about changing list name

File: models/lpjguess/R/model2netcdf.LPJGUESS.R [hunk 3]
```diff
@@ -122,8 +122,8 @@ model2netcdf.LPJGUESS <- function(outdir, sitelat, sitelon, start_date, end_date
     varfile <- file(file.path(outdir, paste(y, "nc", "var", sep = ".")), "w")
     for (i in seq_along(var)) {
       # print(i)
-      ncdf4::ncvar_put(nc, var[[i]], output[[i]])
-      cat(paste(var[[i]]$name, var[[i]]$longname), file = varfile, sep = "\n")
+      ncdf4::ncvar_put(nc, nc_var[[i]], output[[i]])
```
Comment (mdietze) [inline models/lpjguess/R/model2netcdf.LPJGUESS.R:125]: also need to change the nc_create and for above

File: models/maespa/R/model2netcdf.MAESPA.R [hunk 4]
```diff
@@ -71,20 +71,20 @@ model2netcdf.MAESPA <- function(outdir, sitelat, sitelon, start_date, end_date,
...
+    nc_var[[5]] <- PEcAn.utils::to_ncvar("Qle", dims)
     
     ### Output netCDF data
     nc <- ncdf4::nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
     varfile <- file(file.path(outdir, paste(y, "nc", "var", sep = ".")), "w")
     for (i in seq_along(var)) {
       # print(i)
-      ncdf4::ncvar_put(nc, var[[i]], output[[i]])
-      cat(paste(var[[i]]$name, var[[i]]$longname), file = varfile, sep = "\n")
+      ncdf4::ncvar_put(nc, nc_var[[i]], output[[i]])
```
Comment (mdietze) [inline models/maespa/R/model2netcdf.MAESPA.R:86]: nc_create and for

File: models/preles/R/runPRELES.jobsh.R [hunk 5]
```diff
@@ -174,18 +174,18 @@ runPRELES.jobsh <- function(met.file, outdir, parameters, sitelat, sitelon, star
...
+    nc_var[[3]] <- PEcAn.utils::to_ncvar("SoilMoist", dims)
+    nc_var[[4]] <- PEcAn.utils::to_ncvar("Evap", dims)
+    nc_var[[5]] <- PEcAn.utils::to_ncvar("TVeg", dims)
     
     nc <- ncdf4::nc_create(file.path(outdir, paste(y, "nc", sep = ".")), var)
     varfile <- file(file.path(outdir, paste(y, "nc", "var", sep = ".")), "w")
     for (i in seq_along(var)) {
-      ncdf4::ncvar_put(nc, var[[i]], output[[i]])
-      cat(paste(var[[i]]$name, var[[i]]$longname), file = varfile, sep = "\n")
+      ncdf4::ncvar_put(nc, nc_var[[i]], output[[i]])
```
Comment (mdietze) [inline models/preles/R/runPRELES.jobsh.R:187]: nc_create and for

**Review by mdietze — CHANGES_REQUESTED** [review]

**Comment by moki1202** [comment]
@mdietze will be looking at the suggested changes ASAP!

**Review by mdietze — APPROVED** [review]

**Pull request merged**